### ISSUE TITLE
Type hinted class member not visible in inherited class

### ISSUE THREAD

**Issue by spajak** [PyCQA/pylint#4056]
Type hinted class member not visible in inherited class

### Steps to reproduce

```py
from abc import ABCMeta, abstractmethod

class Foo(metaclass=ABCMeta):
    x: str
    y = 1
    @abstractmethod
    def bb(self):
        pass

class Bar(Foo):
    x = 'bbb'
    def bb(self):
        print(self.x)

b = Bar()
b.bb()
```

### Current behavior

```
E1101: Instance of 'Bar' has no 'x' member; maybe 'y'? (no-member)
```

### Expected behavior

No Errors

### pylint --version output

pylint 2.6.0
astroid 2.4.2
Python 3.8.6 (tags/v3.8.6:db45529, Sep 23 2020, 15:52:53) [MSC v.1927 64 bit (AMD64)]

**Comment by Pierre-Sassoulas** [comment]
Thank you for creating the issue, I can reproduce this.

**Comment by cdce8p** [comment]
This issue seems to be fixed when testing with the latest pylint version.
If you still encounter issues, please report back and I'll reopen it.
